coresdk/src/coresdk/database.h

```diff
@@ -1,60 +1,301 @@
+ *
+ *   - show the success or failure of a query using `query_success`,
+ *
+ *   - stores the data from queries which return data. This data is
+ *     accessed using `query_column_for_bool`, `query_column_for_int`,
+ *     `query_column_for_string` and `query_column_for_double`.
+ *
+ * @attribute class query_result
+ */
 typedef struct sk_query_result *query_result;
 
+
+/**
+ * Determines if SplashKit has a database loaded for the supplied name.
+ * This checks against all databases loaded, those loaded without a name
+ * are assigned the filename as a default. If this returns `false`, you may
+ * want to use `load_database` to load in a specific database and give
+ * it the desired name.
+ *
+ * @param name The name to check if a database is loaded.
+ *
+ * @returns Returns `true` if a database with the given `name` has
+ * has been loaded.
+ *
+ * @attribute class   database
+ * @attribute method  has_database
+ */
 bool has_database(string name);
 
+/**
+ * Retrieves a `database` that has been loaded into SplashKit.
+ *
+ * @param name The name of the database to return.
+ *
+ * @returns Returns the `database` that has been loaded with the specified
+ *          `name` using `open_database`.
+ *
+ * @attribute class   database
+ * @attribute method  database_named
+ */
 database database_named(string name);
 
+/**
+ * Calculates the number of row changed in the database at the last time
+ * a changing query was performed on the given database.
+ *
+ * @param db The database to check how many rows changed.
+ *
+ * @returns Returns the `int` which represents how many rows were changed
+ *          in the database.
+ *
+ * @attribute class   database
+ * @attribute method  rows_changed
+ */
 int rows_changed(database db);
 
+/**
+ * @brief Loads or creates, and returns a database
+ *
+ * The supplied `filename` is the name of the file of the database. The
+ * supplied `name` indicates the name to use to refer to this `database`.
+ * The `database` can then be retrieved by passing this `name` to
+ * the `database_named` function.
+ *
+ * @param name      The name used to refer to the database.
+ * @param filename  The filename used to locate the database to use.
+ *
+ * @returns A new `database` with the initialised values provided.
+ *
+ * @attribute class       database
+ * @attribute constructor true
+ */
 database open_database(string name, string filename);
 
+/**
+ * Returns the `query_result` for the query passed into `sql`.
+ *
+ * @param db The database to perform `sql` on.
+ *
+ * @param sql The sql statement to perform on `db`.
+ *
+ * @returns Returns the `query_result` which represents
+ * the result of perfoming `sql` on `db`.
+ *
+ * @attribute class   database
+ * @attribute method  run_sql_on_db
+ */
 query_result run_sql(database db, string sql);
 
+/**
+ * Returns the `query_result` for the query passed into `sql`.
+ *
+ * @param database_name The string name of the database to perform `sql` on.
+ *
```

Comment:
Ditto for the other methods too... I'd just do a regex search to remove these :)